### PR TITLE
Sum normalized functions

### LINKED ISSUE #71: Add functions from Vector3 to Vector2 and Vector4
**Original report ([archived issue](https://osrf-migration.github.io/ignition-gh-pages/#!/ignitionrobotics/ign-math/issues/71)) by Louise Poubel (Bitbucket: [chapulina](https://bitbucket.org/%7B5cfa2075-477b-4ded-bdb9-8d2479544ec4%7D/), GitHub: [chapulina](https://github.com/chapulina)).**

----------------------------------------
For example, there is a [Vector3::Normalized](https://github.com/ignitionrobotics/ign-math/blob/05458c89f0434e2774f19995e713ed7dd94f1018/include/ignition/math/Vector3.hh#L143), but no such function for [Vector2](https://github.com/ignitionrobotics/ign-math/blob/05458c89f0434e2774f19995e713ed7dd94f1018/include/ignition/math/Vector2.hh) or [Vector4](https://github.com/ignitionrobotics/ign-math/blob/05458c89f0434e2774f19995e713ed7dd94f1018/include/ignition/math/Vector4.hh). The same goes for `Round`, `Rounded` and other functions.

IssueForNewDevelopers

### PR DESCRIPTION
**Pull request by luccosta**
Resolves #71 .

This implements the functions Sum() and Normalized() in Vector4 API.

It enables:

```
T sumResult = vec4.Sum();
Vector4 normalVec = vec4.Normalized();
```

### CODE REVIEW

**Comment by luccosta** [comment]
I used the GitHub editor to resolve the conflicts and commit the merge, but CIs `ignition_math-ci-pr_any-homebrew-amd64` and `ignition_math-ci-pr_any-ubuntu_auto-amd64` are not finishing.

**Comment by luccosta** [comment]
@mjcarroll conflict resolved.

**Comment by mjcarroll** [comment]
Failing Bionic CI is currently expected because of ongoing infrastructure improvements.

**Comment by luccosta** [comment]
@mjcarroll is normal that `ignition_math-abichecker-any_to_any-ubuntu_auto-amd64` and `ignition_math-ci-pr_any-ubuntu_auto-amd64` still running?

**Comment by mjcarroll** [comment]
> @mjcarroll is normal that ignition_math-abichecker-any_to_any-ubuntu_auto-amd64 and ignition_math-ci-pr_any-ubuntu_auto-amd64 still running?

Not necessarily "normal", but you can check the build queue at build.osrfoundation.org, you can see that it is a bit backed up at the moment.

File: src/Vector4_TEST.cc [hunk 1]
```diff
@@ -57,6 +57,10 @@ TEST(Vector4dTest, Vector4d)
   v.Normalize();
   EXPECT_EQ(v, math::Vector4d(0.182574, 0.365148, 0.547723, 0.730297));
 
+  // ::Normalized
+  v.Set(1, 2, 3, 4);
+  EXPECT_EQ(v.Normalized(), math::Vector4d(0.182574, 0.365148, 0.547723, 0.730297));
```
Comment (chapulina) [inline src/Vector4_TEST.cc:62]: The code checker is complaining that this line is too long:

`./src/Vector4_TEST.cc:62:  Lines should be <= 80 characters long  [whitespace/line_length] [2]`

You can see the results here: https://github.com/ignitionrobotics/ign-math/pull/140/checks?check_run_id=932511831

You can run the code checker locally with:

```
cd <path to ign-math>
bash tools/code_check.sh
```

Comment (luccosta) [reply src/Vector4_TEST.cc]: Forget to use code checker. To resolve I just break the line putting the second argument under the first like in Style Guide.

**Pull request merged**